engine/services/download_service.cc

```diff
@@ -374,6 +374,9 @@ void DownloadService::ProcessTask(DownloadTask& task, int worker_id) {
       CTL_ERR("Failed to init curl!");
       return;
     }
+    if (!std::filesystem::exists(item.localPath.parent_path())) {
```

Comment:
I think the user who use download service will need to ensure the parent folder to be exists. Download service should not worry about create directory.